src/MICore/Transports/LocalUnixTerminalTransport.cs

```diff
@@ -64,24 +56,38 @@ public override void InitStreams(LaunchOptions options, out StreamReader reader,
             TerminalLauncher terminal = TerminalLauncher.MakeTerminal("DebuggerTerminal", launchDebuggerCommand, localOptions.Environment);
             terminal.Launch(debuggeeDir);
 
+            int shellPid = -1;
+
             using (StreamReader pidReader = new StreamReader(pidStream, Encoding.UTF8, true, UnixUtilities.StreamBufferSize))
             {
-                Task<string> readPidTask = pidReader.ReadLineAsync();
-                if (readPidTask.Wait(TimeSpan.FromSeconds(10)))
+                Task<string> readShellPidTask = pidReader.ReadLineAsync();
+                if (readShellPidTask.Wait(TimeSpan.FromSeconds(5)))
```

Comment:
Sometimes things take *much* longer than you would expect, so short timeouts can be really dangerous. I would leave this at 10.